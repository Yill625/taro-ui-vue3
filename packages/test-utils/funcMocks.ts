import * as utils from "@taro-ui-vue3/utils/common"
import { pxTransformMockFn, getEnvMockFn } from './@tarojs/taro'

let uuid: jest.SpyInstance
let getEnvs: jest.SpyInstance
let pxTransform: jest.SpyInstance

function initMocks() {
  beforeEach(() => {
    jest.mock('@taro-ui-vue3/utils/common', () => {
      return utils
    })

    pxTransform = jest
      .spyOn(utils, 'pxTransform')
      .mockImplementation(pxTransformMockFn)

    uuid = jest
      .spyOn(utils, 'uuid')
      .mockReturnValue('__tests__2021')

    getEnvs = jest
      .spyOn(utils, 'getEnvs')
      .mockImplementation(() => {
        const env = getEnvMockFn()
        return {
          isWEAPP: env === 'WEAPP',
          isALIPAY: env === 'ALIPAY',
          isWEB: env === 'WEB'
        }
      })
  })

  afterEach(() => {
    uuid.mockRestore()
    getEnvs.mockRestore()
    pxTransform.mockRestore()
  })
}

export default initMocks