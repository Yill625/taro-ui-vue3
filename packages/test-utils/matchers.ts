import * as utils from "@taro-ui-vue3/utils/common"

jest.mock('@taro-ui-vue3/utils/common')

let warn: jest.SpyInstance
let uuid: jest.SpyInstance

const asserted: Set<string> = new Set()
const NOOP = () => { }

// Ensure console.warn is called with expected message
const toHaveBeenWarned: jest.CustomMatcher = (received: string) => {
  asserted.add(received)
  const passed = warn.mock.calls.some(args => args[0].indexOf(received) > -1)
  if (passed) {
    return {
      pass: true,
      message: () => `expected "${received}" not to have been warned.`
    }
  } else {
    const msgs = warn.mock.calls.map(args => args[0]).join('\n - ')
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned` +
        (msgs.length
          ? `.\n\nActual messages:\n\n - ${msgs}`
          : ` but no warning was recorded.`)
    }
  }
}

// Ensure previous console.warn is called with expected message
const toHaveBeenWarnedLast: jest.CustomMatcher = (received: string) => {
  asserted.add(received)
  const passed =
    warn.mock.calls[warn.mock.calls.length - 1][0].indexOf(received) > -1
  if (passed) {
    return {
      pass: true,
      message: () => `expected "${received}" not to have been warned last.`
    }
  } else {
    const msgs = warn.mock.calls.map(args => args[0]).join('\n - ')
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned last.\n\nActual messages:\n\n - ${msgs}`
    }
  }
}

// Ensure console.warn is called n times with expected message
const toHaveBeenWarnedTimes: jest.CustomMatcher = (received: string, n: number) => {
  asserted.add(received)
  let found = 0
  warn.mock.calls.forEach(args => {
    if (args[0].indexOf(received) > -1) {
      found++
    }
  })

  if (found === n) {
    return {
      pass: true,
      message: () => `expected "${received}" to have been warned ${n} times.`
    }
  } else {
    return {
      pass: false,
      message: () =>
        `expected "${received}" to have been warned ${n} times but got ${found}.`
    }
  }
}

function initWarnMatchers() {
  beforeEach(() => {
    asserted.clear()
    warn = jest.spyOn(console, 'warn').mockImplementation(NOOP)
    uuid = jest.spyOn(utils, 'uuid').mockReturnValue('__tests__2021')

    // NOT working in concurrent mode
    // should avoid use in test.concurrent()
    expect.extend({
      toHaveBeenWarned,
      toHaveBeenWarnedLast,
      toHaveBeenWarnedTimes
    })
  })

  afterEach(() => {
    const assertedArray = Array.from(asserted)
    const nonAssertedWarnings = warn.mock.calls
      .map(args => args[0])
      .filter(received => {
        return !assertedArray.some(assertedMsg => {
          return received.indexOf(assertedMsg) > -1
        })
      })

    uuid.mockRestore()
    warn.mockRestore()

    if (nonAssertedWarnings.length) {
      throw new Error(
        `test case threw unexpected warnings:\n - ${nonAssertedWarnings.join(
          '\n - '
        )}`
      )
    }
  })
}

export default initWarnMatchers