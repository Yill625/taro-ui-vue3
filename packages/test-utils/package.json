{
  "name": "@taro-ui-vue3/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "license": "MIT",
  "dependencies": {
    "vue": "^3.2.0",
    "@vue/test-utils": "^2.0.0-rc.0"
  }
}