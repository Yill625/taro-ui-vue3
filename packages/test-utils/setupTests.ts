import toHaveBeenWarnedInit from './been-warned'

toHaveBeenWarnedInit()