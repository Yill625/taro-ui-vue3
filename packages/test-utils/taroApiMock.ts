import * as utils from "@taro-ui-vue3/utils/common"
import Taro from '@tarojs/taro'
import { createAnimation } from '@tarojs/taro-h5'

import {
  getEnvMockFn,
  pxTransformMockFn,
  chooseImageMockFn,
  querySelectorMockFn
} from './@tarojs/taro'

Taro.pxTransform = pxTransformMockFn
Taro.chooseImage = chooseImageMockFn
Taro.createSelectorQuery = querySelectorMockFn()
Taro.createAnimation = createAnimation

let uuid: jest.SpyInstance
let getEnvs: jest.SpyInstance
let getenv: jest.SpyInstance
let pxTransform: jest.SpyInstance
let chooseImage: jest.SpyInstance

function initTaroMockApis() {
  beforeEach(() => {
    jest.mock('@tarojs/taro', () => {
      return Taro
    })

    jest.mock('@taro-ui-vue3/utils/common', () => {
      return utils
    })

    getenv = jest
      .spyOn(Taro, 'getEnv')
      .mockImplementation(getEnvMockFn)

    chooseImage = jest
      .spyOn(Taro, 'chooseImage')
      .mockImplementation(chooseImageMockFn)

    pxTransform = jest
      .spyOn(utils, 'pxTransform')
      .mockImplementation(pxTransformMockFn)

    uuid = jest
      .spyOn(utils, 'uuid')
      .mockReturnValue('__tests__2021')

    getEnvs = jest
      .spyOn(utils, 'getEnvs')
      .mockImplementation(() => {
        const env = getEnvMockFn()
        return {
          isWEAPP: env === 'WEAPP',
          isALIPAY: env === 'ALIPAY',
          isWEB: env === 'WEB'
        }
      })
  })

  afterEach(() => {
    uuid.mockRestore()
    getEnvs.mockRestore()
    getenv.mockRestore()
    pxTransform.mockRestore()
    chooseImage.mockRestore()
  })
}

export default initTaroMockApis
