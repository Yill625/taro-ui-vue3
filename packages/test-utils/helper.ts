import { mount, shallowMount, DOMWrapper, VueWrapper } from '@vue/test-utils'
import { VNode, Slot } from '@vue/runtime-core'

export const sleep = async (timeout: number): Promise<null> =>
  new Promise((resolve) => setTimeout(resolve, timeout))

export interface Slots {
  default: Array<string | VNode> | Slot
  [x: string]: Array<string | VNode> | Slot
}

export function mountFactory(
  comp: any,
  components?: Record<string, object>,
  props?: Record<string, any>,
  slots?: Slots
) {
  return mount<any>(comp, {
    global: {
      components
    },
    props,
    slots,
  })
}

export function genMountFn(
  comp: any,
  components?: Record<string, object>,
  useShallowMount?: boolean
) {
  return (props = {}, slots?: Slots) => {
    return useShallowMount
      ? shallowMountFactory(comp, components, props, slots)
      : mountFactory(comp, components, props, slots)
  }
}

export function shallowMountFactory(
  comp: any,
  components?: Record<string, object>,
  props?: Record<string, any>,
  slots?: Slots
) {
  return shallowMount<any>(comp, {
    global: {
      components
    },
    props,
    slots,
  })
}

export interface Dimension {
  top?: number
  bottom?: number
  left?: number
  right?: number
  width?: number
  height?: number
}

export function genDelayedSelectorSpy(
  utils: any,
  dimensions: Dimension
): jest.SpyInstance {
  return jest
    .spyOn(utils, 'delayQuerySelector')
    .mockImplementation((_, sel: string, delay?: number) => {
      return new Promise((resolve) => {
        resolve([dimensions])
      })
    })
}

export async function triggerTouchEvents(
  el: DOMWrapper<Element>,
  touchstartDetails: any,
  touchendDetails: any
) {
  const startDetails = {
    touches: [touchstartDetails],
    changedTouches: [touchstartDetails]
  }

  const endDetails = {
    touches: [touchendDetails],
    changedTouches: [touchendDetails]
  }

  await el.trigger('touchstart', startDetails)
  await el.trigger('touchmove', endDetails)
  await el.trigger('touchend', endDetails)
}

export function testPropClassAndStyle(
  mountFn: (props?: {}, slots?: Slots | undefined) => VueWrapper<any>
) {

  it.concurrent('should render prop class', async () => {
    const wrapper = mountFn({ class: 'test' })
    expect(
      wrapper.classes()
    ).toContain('test')
  })

  it.concurrent('should render prop style', async () => {
    const wrapper = mountFn({ style: { color: 'red' } })
    expect(
      wrapper.attributes('style')
    ).toContain('color: red;')
  })
}

export function hexToRGBA(hex: string, opacity = 1) {
  hex = hex.replace('#', '')
  const rgb = hex
    .match(new RegExp('(.{' + hex.length / 3 + '})', 'g'))!
    .map((l) => parseInt(hex.length % 2 ? l + l : l, 16))

  const rgba = rgb.concat(isFinite(opacity) ? opacity : 1)

  return opacity === 1
    ? `rgb(${rgb.join(', ')})`
    : `rgba(${rgba.join(', ')})`
}