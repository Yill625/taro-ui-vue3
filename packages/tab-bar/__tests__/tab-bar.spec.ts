import { mountFactory, Slots } from '@taro-ui-vue3/test-utils/helper'
import { ref } from 'vue'
import AtTabBar from '../index'

const factory = (values = {}, slots: Slots = { default: [''] }) => {
  return mountFactory(AtTabBar, {}, values, slots)
}

const tabList = [
  { title: '待办事项', iconType: 'bullet-list', text: 'new' },
  { title: '拍照', iconType: 'camera' },
  { title: '文件夹', iconType: 'folder', text: '100', max: '99' },
  {
    title: '领取中心',
    image:
      'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png',
    selectedImage:
      'https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png',
    text: 'new',
  },
]

describe('AtTabBar', () => {

  it('should render default AtTabBar', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- backgroundColor', () => {
    const wrapper = factory({ fixed: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- tabList', () => {
    const wrapper = factory({ tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- iconSize', () => {
    const wrapper = factory({ iconSize: '26', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- fontSize', () => {
    const wrapper = factory({ fontSize: 26, tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- color', () => {
    const wrapper = factory({ color: 'red', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- selectedColor', () => {
    const wrapper = factory({ selectedColor: 'red', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should render prop -- scroll', () => {
    const wrapper = factory({ scroll: true, tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtTabBar Behavior', () => {
  it('should trigger onClick event, and prop current should be equal to index of the item clicked', async () => {
    const current = ref(0)
    const onClick = jest.fn((index) => {
      current.value = index
    })
    const wrapper = factory({ current: current.value, tabList: tabList, onClick: onClick })

    await wrapper.find('.at-tab-bar__item:nth-child(3)').trigger('tap')
    expect(onClick).toBeCalled()

    await wrapper.vm.$nextTick()
    expect(current.value).toBe(2)
  })
})
