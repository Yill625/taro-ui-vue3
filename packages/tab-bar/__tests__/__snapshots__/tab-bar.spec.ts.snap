// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabBar should render default AtTabBar 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
/>
`;

exports[`AtTabBar should render prop -- backgroundColor 1`] = `
<view
  class="at-tab-bar at-tab-bar--fixed"
  style="background-color: rgb(255, 255, 255);"
/>
`;

exports[`AtTabBar should render prop -- class 1`] = `
<view
  class="test at-tab-bar test"
  style="background-color: rgb(255, 255, 255);"
/>
`;

exports[`AtTabBar should render prop -- color 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: rgb(97, 144, 232);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: rgb(97, 144, 232); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: red;"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: red; font-size: 24px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: red;"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: red; font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: red;"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 24px; height: 24px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 24px; height: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;

exports[`AtTabBar should render prop -- fontSize 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: rgb(97, 144, 232);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: rgb(97, 144, 232); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 26px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 26px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 26px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 24px; height: 24px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 24px; height: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 26px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;

exports[`AtTabBar should render prop -- iconSize 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: rgb(97, 144, 232);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: rgb(97, 144, 232); font-size: 26px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: rgb(51, 51, 51); font-size: 26px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: rgb(51, 51, 51); font-size: 26px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 26px; height: 26px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 26px; height: 26px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;

exports[`AtTabBar should render prop -- scroll 1`] = `
<view
  class="at-tab-bar"
  scroll="true"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: rgb(97, 144, 232);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: rgb(97, 144, 232); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 24px; height: 24px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 24px; height: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;

exports[`AtTabBar should render prop -- selectedColor 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: red;"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: red; font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 24px; height: 24px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 24px; height: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;

exports[`AtTabBar should render prop -- style 1`] = `
<view
  class="at-tab-bar"
  style="color: red; background-color: rgb(255, 255, 255);"
/>
`;

exports[`AtTabBar should render prop -- tabList 1`] = `
<view
  class="at-tab-bar"
  style="background-color: rgb(255, 255, 255);"
>
  <view
    class="at-tab-bar__item at-tab-bar__item--active"
    style="color: rgb(97, 144, 232);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-bullet-list"
          style="color: rgb(97, 144, 232); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          待办事项
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-camera"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <!---->
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          拍照
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <text
          class="at-icon at-icon-folder"
          style="color: rgb(51, 51, 51); font-size: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        99+
      </view>
    </view>
    <!---->
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          文件夹
        </view>
        
        <!---->
      </view>
    </view>
  </view>
  <view
    class="at-tab-bar__item"
    style="color: rgb(51, 51, 51);"
  >
    <!---->
    <view
      class="at-badge"
    >
      
      <view
        class="at-tab-bar__icon"
      >
        <image
          class="at-tab-bar__inner-img at-tab-bar__inner-img--inactive"
          mode="widthFix"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 24px; height: 24px;"
        />
        <image
          class="at-tab-bar__inner-img"
          mode="widthFix"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 24px; height: 24px;"
        />
      </view>
      
      <view
        class="at-badge__num"
      >
        new
      </view>
    </view>
    <view>
      <view
        class="at-badge"
      >
        
        <view
          class="at-tab-bar__title"
          style="font-size: 14px;"
        >
          领取中心
        </view>
        
        <!---->
      </view>
    </view>
  </view>
</view>
`;
